x/mint/types/keys.go

```diff
@@ -9,6 +9,18 @@ const (
 	QuerierRoute = ModuleName
 )
 
+// KV Store
 var (
 	LastBlockInfoPrefix = []byte{0x00}
 )
+
+// Transient Store
+var (
+	MintDistribution = []byte{0x00}
+)
+
+// GetMintDistributionRecipientKey gets the store prefix to fetch the inflation distribution for the recipient
+// returns MintDistribution + currentBlockHeight + recipientName
+func GetMintDistributionRecipientKey(blockHeight int64, recipientName string) []byte {
+	return append(append(MintDistribution, byte(blockHeight)), []byte(recipientName)...)
```

Comment:
Historic inflationary rewards might get big over-time. I'm for now to keep as is but let's open an issue. We might want to prune this information every X blocks.

Reply:
Created issue https://github.com/archway-network/archway/issues/313